web/src/message_events.js

```diff
@@ -568,6 +556,20 @@ export function update_messages(events) {
                     moved_message.display_recipient = new_stream_name;
                 }
 
+                // Remove the stream_topic_entry for the old topics;
+                // must be called after we call set message topic since
+                // it calls `get_messages_in_topic` which thinks that
+                // `topic` and `stream` of the messages are correctly set.
+                const num_messages = event_messages.length;
+                if (num_messages > 0) {
+                    stream_topic_history.remove_messages({
+                        stream_id: old_stream_id,
+                        topic_name: orig_topic,
+                        num_messages,
+                        max_removed_msg_id: event_messages[num_messages - 1].id,
+                    });
+                }
```

Comment:
I don't think we want this in a loop? Seems like if it's to remove 3 messages frmo the count (`num_messages`), this code will remove 3 messages 3 times and thus do 9...

So I think a valid option would be to put both this and the `stream_topic_history.add_message` calls outside the loop; or maybe just put this outside the loop just after?

Reply:
Yup, moved it to be called after the loop.